src/interfaces/token.ts

```diff
@@ -121,7 +121,17 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
     const signer = this.ctx.provider.write.getSigner(account);
     if (vault.token === token) {
       const tokenContract = new Contract(token, TokenAbi, signer);
-      return tokenContract.approve(vault.address, amount);
+      const populatedTx = await tokenContract.populateTransaction.approve(vault.address, amount);
```

Comment:
@gotzenx @xgambitox does the FE use this to approve deposits? I can't seem to make this code trigger for me

Reply:
No, token approves we manage it on the FE by ourselves. We execute the transaction